hawkular-inventory-rest-api/src/main/java/org/hawkular/inventory/rest/ResponseUtil.java

```diff
@@ -61,6 +67,20 @@
         return response;
     }
 
+    private static <T> InputStream pageToStream(Page<T> page, ObjectMapper mapper) {
+        PipedInputStream in = new PipedInputStream();
+        new Thread(() -> {
```

Comment:
Wouldn't a `ThreadPoolExecutor` be more performant here instead of starting a new thread every time we page?